software/controller/lib/hal/stepper.cpp

```diff
@@ -15,30 +15,19 @@ limitations under the License.
 
 #include "stepper.h"
 
-#if defined(BARE_STM32)
-
 #include <cmath>
 #include <cstring>
 
 #include "clocks_stm32.h"
 #include "interrupts.h"
 #include "system_timer.h"
 
-// Static data members
-StepMotor StepMotor::motor_[StepMotor::MaxMotors] = {StepMotor(0), StepMotor(1), StepMotor(2),
-                                                     StepMotor(3)};
-// The steppers are chained on the SPI1 bus, which we link with DMA2
-SPI::STM32Channel StepMotor::spi_{SPI::Base::SPI1, DMA::Base::DMA2};
-// The steppers require us to keep CS high for at least 650 ns between bytes.
-// For us, the minimum delay we can control is 1µs
-SPI::DaisyChain</*MaxSlaves=*/StepMotor::MaxMotors, /*MaxRequestsPerSlave=*/10>
-    StepMotor::daisy_chain_{"stepper", "for stepper daisy chain", &spi_,
-                            /*min_cs_high_time=*/microseconds(1)};
+namespace StepMotor {
```

Comment:
Do we have a link to the documentation for this controller that the assumptions come from? Sort of like for the STM HAL?